vdl_tools/shared_tools/embedding_network/embedding_network.py

```diff
@@ -366,41 +429,108 @@ def parse_keyword_response(response_obj):
 
 
 
+def _suffix_from_col_name(colname: str) -> str:
+    m = re.search(r'(_L\d+)$', str(colname))
+    return m.group(1) if m else ""
+
+
+
+def improve_one_sentences(nodesdf,
+                          clusattr='Cluster',
+                          clusname=None,  # Use this for short sentence column name
+                          subject="education",
+                          model="o3-mini",
+                          short_col=None,
+                          long_col=None):
+    """
+        Improves one-sentence summaries for a specific level and REPLACES the short column.
+        - If short_col/long_col are provided, uses them directly.
+        - Else if clusname is provided, infers columns as:
+             short = f"{clusname}{suffix_from(clusattr)}"
+             long  = short + "_long"
+        - Else falls back to: 'clus_sentence_short' / 'clus_sentence_long'
+        """
+    out = nodesdf.copy()
+    for col in (clusattr, short_col, long_col):
+        if col not in out.columns:
+            raise KeyError(f"Expected column '{col}' not found.")
+
+        # Minimal table for prompt
+    df_min = out[[clusattr, short_col, long_col]].drop_duplicates()
+    # print(df.shape)
+    assistant_prompt = define_assistant_prompt(subject)
+    # get the prompt for the cluster
+    review_prompt = format_prompt_review_titles(df_min, clusattr)
+
+    response = oai_utils.CLIENT.chat.completions.create(
+        model=model,
+        messages=[
+            {
+                "role": "system",
+                "content": [{"type": "text", "text": assistant_prompt + "\n"}],
+            },
+            {"role": "user", "content": [{"type": "text", "text": review_prompt}]},
+        ],
+        response_format={"type": "json_object"},
+        temperature=1,
+        max_completion_tokens=10000,
+        top_p=1,
+        frequency_penalty=0,
+        presence_penalty=0,
+        store=False,
+    )
+
+    # get the response json
+    json_resp = json.loads(response.choices[0].message.content)
+    out[short_col] = out[clusattr].map(json_resp)
+    return out
+
+
 
 def get_cluster_sentences_from_text(
     nodesdf,
     textcol,
     clusattr='Cluster',
+    clusname=None,             # Use this for short sentence column name
     subject=None,
     model="gpt-4.1-mini",
+    n_tags = None,
+    n_entities=None,
+    sample_texts=None,
+    improve = False,
+    improve_model="o3-mini"  # Model for improving one-sentence summaries
```

Comment:
```suggestion
    improve_model="o3-mini"  # Model for improving one-sentence summaries
    centrality_col="ClusterCentrality"
```

This is an argument for the sampling function, so I'd make it an argument here to and make this the default